features/207-sdk-api-consistency/record-api.md

```diff
@@ -32,23 +32,44 @@ https://github.com/SkygearIO/features/pull/227 is considered in this doc.
 
 ```objc
 - (void)performQuery:(SKYQuery *)query
-    completionHandler:
-        (void (^_Nullable)(NSArray *_Nullable results, NSError *_Nullable error))completionHandler;
+          completion:
+            (void (^_Nullable)(NSArray *_Nullable results, NSError *_Nullable error))completion;
 
 - (void)performCachedQuery:(SKYQuery *)query
-         completionHandler:(void (^_Nullable)(NSArray *_Nullable results, BOOL pending,
-                                              NSError *_Nullable error))completionHandler;
+                completion:(void (^_Nullable)(NSArray *_Nullable results, BOOL pending,
+                                              NSError *_Nullable error))completion;
 
 - (void)fetchRecordWithType:(NSString *)recordType
                    recordID:(NSString *)recordID
-          completionHandler:(void (^)(SKYRecord *record, NSError *error))completionHandler;
+                 completion:(void (^_Nullable)(SKYRecord *_Nullable record,
+                                               NSError *_Nullable error))completion;
+
+// Generic RecordResult object for fetch, non-atomic save and delete
+@interface SKYRecordResult<ObjectType> : NSObject
+
+@property (nonatomic, readonly) ObjectType value;
+@property (nonatomic, readonly) NSError* error;
+
+@end
```

Comment:
May I clarify that this class will not exist in Swift? Otherwise, the coexistence of `SKYRecordResult` and `RecordResult` would be very confusing.

Reply:
I updated both obj-c and swift using the same name `SKYRecordResult`. In obj-c, I added `NS_SWIFT_NAME` to rename the object and having enum implementation in swift. See if it looks good? @ben181231 @cheungpat 

commit: bb27194

Reply:
👍